en/guides/codablerouting.md

```diff
@@ -15,172 +15,289 @@ redirect_from:
+This guide shows you step by step how to set up Codable routes on a Kitura server. 
 
-```swift
-router.post("/todos", handler: storeHandler)
-```
+## Let's build a Server!
 
-and then provide an implementation of the storeHandler:
+### Getting started
+First we need a Kitura server to add our routes to. Kitura has a [command-line interface](https://github.com/IBM-Swift/kitura-cli) which provides us with utilities for quickly creating a Kitura server.  
+> ![tip] You can find installation instructions for the command-line interface in the [Getting Started](/{{ page.lang }}/starter/gettingstarted.html) section. 
 
-```swift
-func storeHandler(todo: ToDo, completion: (ToDo?, RequestError?) -> Void ) -> Void {
-    todoStore.append(todo)
-    completion(todo, nil)
-}
-```
+Once we've installed the Kitura command-line interface, open a terminal window and run the following:  
 
-Here the store handler accepts a todo item of type `ToDo` and responds using an asynchronous completion handler. The handler then stores the received todo item in the todoStore and returns with the created todo item in the case of success, and an `Error` in the the case of failure.
+1. Create a directory:  
 
-The input parameter and the response parameter in completion handler can be any Swift type you want, as long as it conforms to Codable. For example, we could use the following as our `ToDo` type:
+```
+mkdir ~/CodableRouting
+```
+2. Change to the new directory:  
 
-```swift
-public struct ToDo : Codable {
-    public var title: String        // title of the todo item
-    public var order: Int           // order to display the todo item
-    public var completed: Bool      // has the item been completed
-}
 ```
+cd ~/CodableRouting
+```
+3. Create the Kitura server:  
 
-At this point the server has stored the item, but the client has no way of referring to that specific item on the server if it wants to get, update, or delete that item. In order to do that, some kind of unique identifier is required.
+```
+kitura init
+```
 
-Returning an identifier to the client for the stored data is typically done in one of two ways:
+`kitura init` creates and builds a starter Kitura application and will take a couple of minutes to complete. This includes best-practice implementations of health checking and monitoring as well as configuration files to allow easy deployment to a Docker container, a Kubernetes cluster, or the IBM Cloud.
 
-* ### Adding an additional Optional field to the type
-Here the ToDo struct would be updated to add an extra field:
-```swift
-public struct ToDo : Codable {
-    public var title: String       
-    public var order: Int      
-    public var completed: Bool
-    public var id: Int?         // Additional field for identifier
-}
-```
+`kitura init` also generates an Xcode project.  
```

Comment:
I don't see the change I requested to replace this and the line below with the wording:
"Open the generated `CodableRouting.xcodeproj` project in Xcode:"